### ISSUE TITLE
'-fno-strict-overflow'-warnings litter certain builds on darwin

### ISSUE THREAD

**Issue by knedlsepp** [NixOS/nixpkgs#39687]
## Issue description
When building C++ on darwin, oftentimes the following warning litters the logs:
```
clang-5.0: warning: argument unused during compilation: '-fno-strict-overflow' [-Wunused-command-line-argument]
```

**Comment by knedlsepp** [comment]
This might be a good spot to disable it: https://github.com/NixOS/nixpkgs/blob/ff3fe105f209a5e934b4062a9259e1e0cc587c8e/pkgs/stdenv/generic/make-derivation.nix#L76

**Comment by knedlsepp** [comment]
Hm. It seems the flag *should* actually be accepted by clang. I don't know why these warnings keep turning up though.

**Comment by dtzWill** [comment]
It says unused, not unsupported.  I think it can happen, for example, when using clang to with assembly files and such?

**Comment by knedlsepp** [comment]
Yes, this is mostly cosmetic, I just had a look at the current Cython darwin build (in #39576), which sports an impressive number of 1828 of those warnings: https://nix-cache.s3.amazonaws.com/log/x9ff903kpnmmvf39cdw4pbp08rl30mmh-python3.6-Cython-0.28.1.drv

**Comment by matthewbauer** [comment]
LOL that is a lot. It is definitely not just Nixpkgs that is doing this though (https://github.com/jedisct1/libsodium/issues/201). We can definitely set ```-Wno-unused-command-line-argument``` but that runs the risk of disregarding legitimate cases of that warning.

**Comment by athas** [comment]
This is breaking packages that use `-Werror`.  What's the solution supposed to be?

**Comment by athas** [comment]
To future readers, this is the solution:

```
hardeningDisable = ["strictoverflow"]; 
```

**Comment by vcunat** [comment]
Should be solved generally now, via PR #243595
